baas-node-state/src/main/scala/com/ing/baker/baas/state/ServiceDiscovery.scala

```diff
@@ -1,143 +1,145 @@
 package com.ing.baker.baas.state
 
+import akka.Done
 import akka.actor.ActorSystem
-import akka.stream.scaladsl.{Keep, Sink}
-import akka.stream.{KillSwitches, Materializer}
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.stream.{KillSwitches, Materializer, UniqueKillSwitch}
 import cats.effect.concurrent.Ref
 import cats.effect.{ContextShift, IO, Resource, Timer}
 import cats.implicits._
 import com.ing.baker.baas.interaction.RemoteInteractionClient
 import com.ing.baker.il.petrinet.InteractionTransition
 import com.ing.baker.runtime.akka.internal.InteractionManager
 import com.ing.baker.runtime.scaladsl.InteractionInstance
-import com.typesafe.scalalogging.LazyLogging
+import com.typesafe.scalalogging.{LazyLogging, Logger}
 import org.http4s.Uri
+import org.http4s.client.Client
 import skuber._
 import skuber.api.client.{EventType, KubernetesClient}
 import skuber.json.format._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
 
-object ServiceDiscovery extends LazyLogging {
+object ServiceDiscovery {
 
-  private[state] type RecipeName = String
+  val logger: Logger = Logger("bakery.ServiceDiscovery")
```

Comment:
what is the point of having logger other than class name?

Reply:
I thought it would be simpler to read shorter logger names for well known components, but that could be reconsidered later, we still have to create a story where we review logging in general